### PR TITLE
illumos gcc 14.2.0 bringup

### PR DESCRIPTION
**Pull request by citrus-it**
Import the patches used by OmniOS and OpenIndiana for their gcc-14 packages. Test results to follow.

### CODE REVIEW

File: gcc/cp/module.cc [hunk 1]
```diff
@@ -1675,7 +1675,7 @@ elf_in::defrost (const char *name)
 	      set_error (errno);
 	  else
 	    {
-	      if (madvise (mapping, hdr.pos, MADV_RANDOM))
+	      if (posix_madvise (mapping, hdr.pos, MADV_RANDOM))
```
Comment (rmustacc) [inline gcc/cp/module.cc]: Strictly speaking if we're using `posix_madvise` we should use `POSIX_MADV_RANDOM`. I realize they're defined to be the same value which is why we get away with it. I'm guessing this is to deal with the `madvise` visibility issues in the past prior to  illumos#14418?

**Review by rmustacc — COMMENTED** [review]
Aside from the new set of patches at the end, did anything else major change in this?

File: gcc/cp/module.cc (hunk 1 above)
Comment (citrus-it) [reply gcc/cp/module.cc]: With 14418, we can completely drop this patch. Done.

**Comment by citrus-it** [comment]
> Aside from the new set of patches at the end, did anything else major change in this?

No, apart from the new cmn_err() support for 'j' and 'z' (after 16768), this is the same patch set that we've been carrying in OmniOS and IO for gcc 13, just brought forward to gcc 14.

I'll add issue numbers to the commit messages where I have them and push again.

**Comment by tsoome** [comment]
Nice, thanks!

**Comment by rmustacc** [comment]
> > Aside from the new set of patches at the end, did anything else major change in this?
> 
> No, apart from the new cmn_err() support for 'j' and 'z' (after 16768), this is the same patch set that we've been carrying in OmniOS and IO for gcc 13, just brought forward to gcc 14.

OK, thanks. Let's make sure to bring the j/z to gcc10 as well as we'll need it there as well. I think it'd be good to get that in so we can do the 14/updated 10 crossover at once.

> I'll add issue numbers to the commit messages where I have them and push again.

Sounds good. Once we have test results, which I know are forthcoming and we review them, then we'll approve.

**Comment by citrus-it** [comment]
> OK, thanks. Let's make sure to bring the j/z to gcc10 as well as we'll need it there as well. I think it'd be good to get that in so we can do the 14/updated 10 crossover at once.

Opened https://github.com/illumos/gcc/pull/54 for adding that to gcc10.

**Comment by citrus-it** [comment]
Summary test results:

```
                === gcc Summary for unix//-m32 ===
# of expected passes            193150
# of unexpected failures        301
# of unexpected successes       26
# of expected failures          1547
# of unresolved testcases       75
# of unsupported tests          4800
                === gcc Summary for unix//-m64 ===
# of expected passes            199374
# of unexpected failures        269
# of unexpected successes       19
# of expected failures          1563
# of unsupported tests          3644
                === gcc Summary for unix//-m64/-msave-args ===
# of expected passes            191839
# of unexpected failures        405
# of expected failures          1577
# of unsupported tests          3736
                === gcc Summary ===
# of expected passes            584363
# of unexpected failures        975
# of unexpected successes       45
# of expected failures          4687
# of unresolved testcases       75
# of unsupported tests          12180
```
```
                === g++ Summary for unix//-m32 ===
# of expected passes            245470
# of unexpected failures        162
# of expected failures          2319
# of unresolved testcases       12
# of unsupported tests          11627
                === g++ Summary for unix//-m64 ===
# of expected passes            245922
# of unexpected failures        108
# of expected failures          2324
# of unresolved testcases       4
# of unsupported tests          11528
                === g++ Summary for unix//-m64/-msave-args ===
# of expected passes            245906
# of unexpected failures        124
# of expected failures          2324
# of unresolved testcases       4
# of unsupported tests          11528
                === g++ Summary ===
# of expected passes            737298
# of unexpected failures        394
# of expected failures          6967
# of unresolved testcases       20
# of unsupported tests          34683
```
```
                === libatomic Summary for unix//-m32 ===
# of expected passes            44
# of unsupported tests          5
                === libatomic Summary for unix//-m64 ===
# of expected passes            54
                === libatomic Summary for unix//-m64/-msave-args ===
# of expected passes            54
                === libatomic Summary ===
# of expected passes            152
# of unsupported tests          5
```
```
                === libgomp Summary for unix//-m32 ===
# of expected passes            5494
# of unexpected failures        1
# of expected failures          34
# of unsupported tests          370
                === libgomp Summary for unix//-m64 ===
# of expected passes            5496
# of expected failures          34
# of unsupported tests          369
                === libgomp Summary for unix//-m64/-msave-args ===
# of expected passes            5496
# of expected failures          34
# of unsupported tests          369
                === libgomp Summary ===
# of expected passes            16486
# of unexpected failures        1
# of expected failures          102
# of unsupported tests          1108
```
```
                === libstdc++ Summary for unix//-m32 ===
# of expected passes            18254
# of unexpected failures        3
# of expected failures          127
# of unresolved testcases       1
# of unsupported tests          792
                === libstdc++ Summary for unix//-m64 ===
# of expected passes            18240
# of unexpected failures        5
# of expected failures          127
# of unresolved testcases       1
# of unsupported tests          798
                === libstdc++ Summary for unix//-m64/-msave-args ===
# of expected passes            18240
# of unexpected failures        5
# of expected failures          127
# of unresolved testcases       1
# of unsupported tests          798
                === libstdc++ Summary ===
# of expected passes            54734
# of unexpected failures        13
# of expected failures          381
# of unresolved testcases       3
# of unsupported tests          2388
```

**Comment by rmustacc** [comment]
Thanks for the tests results there @citrus-it. I realize it's hard to compare 10.x and 14.x, but do we have a sense where the increase in unexpected failures/passes is coming from in general? I realize we also have a much larger number of newly increasing passing tests, which is good! I would note that @richlowe's comments on the cmn_err bits probably applies here.

**Comment by citrus-it** [comment]
There are 296 unique failing gcc tests across the 32-bit, 64-bit and 64-bit -msave-args.
A lot of these tests scan the final assembly for patterns, offsets etc. and are confounded by our propensity to insist on preserving the frame pointer or the extra instructions that `-msave-args` produces. 

The number of failures due to the frame pointer is over 100, based on grepping for failures that find things like %rsp when they don't expect it  [example - pr81736-2.c](https://github.com/illumos/gcc/blob/il-14_2_0/gcc/testsuite/gcc.target/i386/pr81736-2.c)).

Similarly for -msave-args there are a large number that fail because they don't expect to see %rsp, or movq or some other instruction that we insert with that option -- [example1](https://github.com/illumos/gcc/blob/il-14_2_0/gcc/testsuite/gcc.target/x86_64/abi/callabi/leaf-1.c) [example2](https://github.com/illumos/gcc/blob/il-14_2_0/gcc/testsuite/gcc.target/i386/pr43618-1.c)

I could work through the list but I think most will fit into one of these categories.

I'll run the testsuite again without the force-frame-pointer patch and see how much things improve.

**Comment by citrus-it** [comment]
Running the testsuite without the frame-pointer patches results in a reduction in the number of failing tests, as expected.

```diff
                === gcc Summary for unix//-m64 ===
-# of unexpected failures        269
+# of unexpected failures        118
```

I looked at the first failing test and that is `isatty-1.c`, testing for errno being set when `isatty()` returns 0. I'm running this test on Helios which does not have [illumos 15220](https://www.illumos.org/issues/15220), so that's expected. This one accounts for four of the failures.

Looking at some more failures, these are using gdb to set breakpoints and very various things in target binaries. Here's one of them:

```
Spawning: gdb -nx -nw -quiet -batch -x pr98219-1.gdb ./pr98219-1.exe
Breakpoint 1 at 0x4010a0: file .../gcc.dg/guality/pr98219-1.c, line 22.
[Thread debugging using libthread_db enabled]
[New Thread 1 (LWP 1)]
[Switching to Thread 1 (LWP 1)]
Thread 2 hit Breakpoint 1, fn (frame=0xfffff9ffffdeedd8, uirrv=18446737476341550272) at ...gcc.dg/guality/pr98219-1.c:23
23        if (UIRRV != uirrv)           /* BREAK */
$1 = 305419888
$2 = 305419889
305419888 != 305419889
FAIL: gcc.dg/guality/pr98219-1.c   -O2  -DPREVENT_OPTIMIZATION  line 22 frame->rip == 0x12345671
```

When I build and run [the test file](https://github.com/illumos/gcc/blob/il-14_2_0/gcc/testsuite/gcc.dg/guality/pr98219-1.c) manually, it does not abort, suggesting that this is a problem with `gdb` on illumos rather than the code produced by gcc. In this case the binary is built with `-mgeneral-regs-only` and for some reason, `gdb` is getting `frame->rip` as 0x12345670 instead of 0x12345671. If it were right, the binary would abort when run.

```
> main::dis
main:                           pushq  $0x12345673
main+5:                         pushq  $0x12345672
main+0xa:                       pushq  $0x12345671
main+0xf:                       pushq  $0x12345670
main+0x14:                      jmp    -0x91    <fn>
main+0x19:                      movl   $0x0,%eax
main+0x1e:                      ret
> ::bp fn
> ::run
mdb: stop at fn
mdb: target stopped at:
fn:             pushq  %rdi
> <rsp::dump -l 20
                    0 1 2 3  4 5 6 7 \/ 9 a b  c d e f  01234567v9abcdef
fffff9ffffdf18c0:  c04c19ef fff9ffff 70563412 00000000  .L......pV4.....
fffff9ffffdf18d0:  71563412 00000000 72563412 00000000  qV4.....rV4.....
fffff9ffffdf18e0:  73563412 00000000 670f4000 00000000  sV4.....g.@.....
> ::print -xaht struct __uintr_frame
0 struct __uintr_frame {
    0 unsigned long long rip
    8 unsigned long long rflags
    10 unsigned long long rsp
}
```
```
% gdb a.out
GNU gdb (GDB) 13.1
(gdb) break fn
Breakpoint 1 at 0x401090: file gcc/testsuite/gcc.dg/guality/pr98219-1.c, line 23.
(gdb) run
Starting program: /home/andy/helios/tmp/omnios-build/gcc-14.2.0/gcc-14.2.0/a.out
[Thread debugging using libthread_db enabled]
[New Thread 1 (LWP 1)]
[Switching to Thread 1 (LWP 1)]

Thread 2 hit Breakpoint 1, fn (frame=0xfffff9ffffdf6718,
    uirrv=18446737476351839424) at gcc/testsuite/gcc.dg/guality/pr98219-1.c:23
23        if (UIRRV != uirrv)           /* BREAK */
(gdb) print /x *frame
$1 = {rip = 0x12345670, rflags = 0x12345671, rsp = 0x12345672}
```

Almost all of the remaining failures are in this `gdb` category. Here's the full list for `gcc -m64`

```
703851:Running target unix//-m64
857476:FAIL: c-c++-common/analyzer/isatty-1.c  (test for warnings, line 32)
857480:FAIL: c-c++-common/analyzer/isatty-1.c  (test for warnings, line 47)
857482:FAIL: c-c++-common/analyzer/isatty-1.c  (test for warnings, line 57)
857483:FAIL: c-c++-common/analyzer/isatty-1.c (test for excess errors)
885868:FAIL: gcc.dg/debug/btf/btf-cvr-quals-1.c scan-assembler-times [\t ]0xb000000[\t ]+[^\n]*btt_info 2
886417:FAIL: gcc.dg/debug/ctf/ctf-cvr-quals-1.c scan-assembler-times [\t ]0[\t ]+[^\n]*ctt_name 7
886421:FAIL: gcc.dg/debug/ctf/ctf-cvr-quals-1.c scan-assembler-times [\t ]0x36000000[\t ]+[^\n]*ctt_info 2
926139:FAIL: gcc.dg/initpri1.c execution test
926172:FAIL: gcc.dg/initpri3.c execution test
957675:FAIL: gcc.dg/initpri1.c execution test
957708:FAIL: gcc.dg/initpri3.c execution test
1032725:FAIL: gcc.dg/guality/pr98219-1.c   -O2  -DPREVENT_OPTIMIZATION  line 22 uirrv == 0x12345670
1032740:FAIL: gcc.dg/guality/pr98219-1.c   -O2  -DPREVENT_OPTIMIZATION  line 22 frame->rip == 0x12345671
1032755:FAIL: gcc.dg/guality/pr98219-1.c   -O2  -DPREVENT_OPTIMIZATION  line 22 frame->rflags == 0x12345672
1032770:FAIL: gcc.dg/guality/pr98219-1.c   -O2  -DPREVENT_OPTIMIZATION  line 22 frame->rsp == 0x12345673
1032796:FAIL: gcc.dg/guality/pr98219-1.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 22 uirrv == 0x12345670
1032811:FAIL: gcc.dg/guality/pr98219-1.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 22 frame->rip == 0x12345671
1032826:FAIL: gcc.dg/guality/pr98219-1.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 22 frame->rflags == 0x12345672
1032841:FAIL: gcc.dg/guality/pr98219-1.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 22 frame->rsp == 0x12345673
1032863:FAIL: gcc.dg/guality/pr98219-1.c   -Os  -DPREVENT_OPTIMIZATION  line 22 uirrv == 0x12345670
1032878:FAIL: gcc.dg/guality/pr98219-1.c   -Os  -DPREVENT_OPTIMIZATION  line 22 frame->rip == 0x12345671
1032898:FAIL: gcc.dg/guality/pr98219-1.c   -Os  -DPREVENT_OPTIMIZATION  line 22 frame->rflags == 0x12345672
1032913:FAIL: gcc.dg/guality/pr98219-1.c   -Os  -DPREVENT_OPTIMIZATION  line 22 frame->rsp == 0x12345673
1032998:FAIL: gcc.dg/guality/pr98219-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 22 uirrv == 0x12345670
1033013:FAIL: gcc.dg/guality/pr98219-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 22 frame->rip == 0x12345671
1033028:FAIL: gcc.dg/guality/pr98219-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 22 frame->rflags == 0x12345672
1033048:FAIL: gcc.dg/guality/pr98219-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 22 frame->rsp == 0x12345673
1033070:FAIL: gcc.dg/guality/pr98219-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 22 uirrv == 0x12345670
1033090:FAIL: gcc.dg/guality/pr98219-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 22 frame->rip == 0x12345671
1033105:FAIL: gcc.dg/guality/pr98219-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 22 frame->rflags == 0x12345672
1033120:FAIL: gcc.dg/guality/pr98219-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 22 frame->rsp == 0x12345673
1033278:FAIL: gcc.dg/guality/pr98219-2.c   -O2  -DPREVENT_OPTIMIZATION  line 34 uirrv == 0x12345670
1033293:FAIL: gcc.dg/guality/pr98219-2.c   -O2  -DPREVENT_OPTIMIZATION  line 34 frame->rip == 0x12345671
1033308:FAIL: gcc.dg/guality/pr98219-2.c   -O2  -DPREVENT_OPTIMIZATION  line 34 frame->rflags == 0x12345672
1033327:FAIL: gcc.dg/guality/pr98219-2.c   -O2  -DPREVENT_OPTIMIZATION  line 34 frame->rsp == 0x12345673
1033349:FAIL: gcc.dg/guality/pr98219-2.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 34 uirrv == 0x12345670
1033368:FAIL: gcc.dg/guality/pr98219-2.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 34 frame->rip == 0x12345671
1033383:FAIL: gcc.dg/guality/pr98219-2.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 34 frame->rflags == 0x12345672
1033398:FAIL: gcc.dg/guality/pr98219-2.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 34 frame->rsp == 0x12345673
1033420:FAIL: gcc.dg/guality/pr98219-2.c   -Os  -DPREVENT_OPTIMIZATION  line 34 uirrv == 0x12345670
1033435:FAIL: gcc.dg/guality/pr98219-2.c   -Os  -DPREVENT_OPTIMIZATION  line 34 frame->rip == 0x12345671
1033450:FAIL: gcc.dg/guality/pr98219-2.c   -Os  -DPREVENT_OPTIMIZATION  line 34 frame->rflags == 0x12345672
1033465:FAIL: gcc.dg/guality/pr98219-2.c   -Os  -DPREVENT_OPTIMIZATION  line 34 frame->rsp == 0x12345673
1033555:FAIL: gcc.dg/guality/pr98219-2.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 34 uirrv == 0x12345670
1033575:FAIL: gcc.dg/guality/pr98219-2.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 34 frame->rip == 0x12345671
1033595:FAIL: gcc.dg/guality/pr98219-2.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 34 frame->rflags == 0x12345672
1033610:FAIL: gcc.dg/guality/pr98219-2.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 34 frame->rsp == 0x12345673
1033632:FAIL: gcc.dg/guality/pr98219-2.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 34 uirrv == 0x12345670
1033651:FAIL: gcc.dg/guality/pr98219-2.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 34 frame->rip == 0x12345671
1033671:FAIL: gcc.dg/guality/pr98219-2.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 34 frame->rflags == 0x12345672
1033691:FAIL: gcc.dg/guality/pr98219-2.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 34 frame->rsp == 0x12345673
1037770:FAIL: gcc.dg/guality/pr54200.c  -Og -DPREVENT_OPTIMIZATION  line 20 z == 3
1038501:FAIL: gcc.dg/guality/pr54519-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 20 y == 25
1038521:FAIL: gcc.dg/guality/pr54519-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 20 z == 6
1038550:FAIL: gcc.dg/guality/pr54519-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 23 y == 117
1038565:FAIL: gcc.dg/guality/pr54519-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 23 z == 8
1038605:FAIL: gcc.dg/guality/pr54519-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 20 y == 25
1038620:FAIL: gcc.dg/guality/pr54519-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 20 z == 6
1038654:FAIL: gcc.dg/guality/pr54519-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 23 y == 117
1038669:FAIL: gcc.dg/guality/pr54519-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 23 z == 8
1039044:FAIL: gcc.dg/guality/pr54519-2.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 17 y == 25
1039103:FAIL: gcc.dg/guality/pr54519-2.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 17 y == 25
1039769:FAIL: gcc.dg/guality/pr54519-3.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 20 y == 25
1039784:FAIL: gcc.dg/guality/pr54519-3.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 20 z == 6
1039817:FAIL: gcc.dg/guality/pr54519-3.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 23 y == 117
1039837:FAIL: gcc.dg/guality/pr54519-3.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 23 z == 8
1039877:FAIL: gcc.dg/guality/pr54519-3.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 20 y == 25
1039896:FAIL: gcc.dg/guality/pr54519-3.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 20 z == 6
1039925:FAIL: gcc.dg/guality/pr54519-3.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 23 y == 117
1039940:FAIL: gcc.dg/guality/pr54519-3.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 23 z == 8
1040279:FAIL: gcc.dg/guality/pr54519-4.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 17 y == 25
1040334:FAIL: gcc.dg/guality/pr54519-4.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 17 y == 25
1040958:FAIL: gcc.dg/guality/pr43051-1.c   -O3 -fomit-frame-pointer -funroll-loops -fpeel-loops -ftracer -finline-functions  -DPREVENT_OPTIMIZATION  line 34 c == &a[0]
1041011:FAIL: gcc.dg/guality/pr43051-1.c   -O3 -fomit-frame-pointer -funroll-loops -fpeel-loops -ftracer -finline-functions  -DPREVENT_OPTIMIZATION  line 39 c == &a[0]
1045641:FAIL: gcc.dg/guality/pr68037-1.c   -O2  -DPREVENT_OPTIMIZATION  line 33 error == 0x12345670
1045656:FAIL: gcc.dg/guality/pr68037-1.c   -O2  -DPREVENT_OPTIMIZATION  line 33 frame->ip == 0x12345671
1045676:FAIL: gcc.dg/guality/pr68037-1.c   -O2  -DPREVENT_OPTIMIZATION  line 33 frame->cs == 0x12345672
1045691:FAIL: gcc.dg/guality/pr68037-1.c   -O2  -DPREVENT_OPTIMIZATION  line 33 frame->flags == 0x12345673
1045706:FAIL: gcc.dg/guality/pr68037-1.c   -O2  -DPREVENT_OPTIMIZATION  line 33 frame->sp == 0x12345674
1045721:FAIL: gcc.dg/guality/pr68037-1.c   -O2  -DPREVENT_OPTIMIZATION  line 33 frame->ss == 0x12345675
1045743:FAIL: gcc.dg/guality/pr68037-1.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 33 error == 0x12345670
1045758:FAIL: gcc.dg/guality/pr68037-1.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 33 frame->ip == 0x12345671
1045773:FAIL: gcc.dg/guality/pr68037-1.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 33 frame->cs == 0x12345672
1045788:FAIL: gcc.dg/guality/pr68037-1.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 33 frame->flags == 0x12345673
1045803:FAIL: gcc.dg/guality/pr68037-1.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 33 frame->sp == 0x12345674
1045823:FAIL: gcc.dg/guality/pr68037-1.c   -O3 -g  -DPREVENT_OPTIMIZATION  line 33 frame->ss == 0x12345675
1045850:FAIL: gcc.dg/guality/pr68037-1.c   -Os  -DPREVENT_OPTIMIZATION  line 33 error == 0x12345670
1045870:FAIL: gcc.dg/guality/pr68037-1.c   -Os  -DPREVENT_OPTIMIZATION  line 33 frame->ip == 0x12345671
1045885:FAIL: gcc.dg/guality/pr68037-1.c   -Os  -DPREVENT_OPTIMIZATION  line 33 frame->cs == 0x12345672
1045905:FAIL: gcc.dg/guality/pr68037-1.c   -Os  -DPREVENT_OPTIMIZATION  line 33 frame->flags == 0x12345673
1045920:FAIL: gcc.dg/guality/pr68037-1.c   -Os  -DPREVENT_OPTIMIZATION  line 33 frame->sp == 0x12345674
1045935:FAIL: gcc.dg/guality/pr68037-1.c   -Os  -DPREVENT_OPTIMIZATION  line 33 frame->ss == 0x12345675
1046048:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 33 error == 0x12345670
1046063:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 33 frame->ip == 0x12345671
1046078:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 33 frame->cs == 0x12345672
1046093:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 33 frame->flags == 0x12345673
1046108:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 33 frame->sp == 0x12345674
1046127:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 33 frame->ss == 0x12345675
1046149:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 33 error == 0x12345670
1046169:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 33 frame->ip == 0x12345671
1046188:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 33 frame->cs == 0x12345672
1046203:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 33 frame->flags == 0x12345673
1046218:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 33 frame->sp == 0x12345674
1046233:FAIL: gcc.dg/guality/pr68037-1.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 33 frame->ss == 0x12345675
1054720:FAIL: gcc.dg/guality/pr54519-5.c   -O2 -flto -flto-partition=none  -DPREVENT_OPTIMIZATION line 17 y == 25
1054779:FAIL: gcc.dg/guality/pr54519-5.c   -O2 -flto  -DPREVENT_OPTIMIZATION line 17 y == 25
1054978:FAIL: gcc.dg/guality/pr54519-6.c   -Os  -DPREVENT_OPTIMIZATION  line 11 x == 2
1061308:FAIL: gcc.dg/guality/pr56154-1.c  -Og -DPREVENT_OPTIMIZATION  line pr56154-1.c:20 x.a == 6
1068900:FAIL: gcc.dg/guality/pr59776.c  -Og -DPREVENT_OPTIMIZATION  line pr59776.c:17 s1.f == 5.0
1068915:FAIL: gcc.dg/guality/pr59776.c  -Og -DPREVENT_OPTIMIZATION  line pr59776.c:17 s1.g == 6.0
1068930:FAIL: gcc.dg/guality/pr59776.c  -Og -DPREVENT_OPTIMIZATION  line pr59776.c:17 s2.f == 0.0
1068969:FAIL: gcc.dg/guality/pr59776.c  -Og -DPREVENT_OPTIMIZATION  line pr59776.c:20 s1.f == 5.0
1068984:FAIL: gcc.dg/guality/pr59776.c  -Og -DPREVENT_OPTIMIZATION  line pr59776.c:20 s1.g == 6.0
1069004:FAIL: gcc.dg/guality/pr59776.c  -Og -DPREVENT_OPTIMIZATION  line pr59776.c:20 s2.f == 5.0
1072485:FAIL: gcc.dg/ipa/ipa-icf-38.c scan-ltrans-tree-dump-not optimized "Function bar"
```

**Comment by rmustacc** [comment]
Thanks! This has landed and been tagged as gcc-14.2.0-il-1. I did push this in manually to preserve the chain's history and commits.

**Pull request closed without merging**